CRM/SMS/Provider.php

```diff
@@ -207,9 +209,10 @@ public function processInbound($from, $body, $to = NULL, $trackID = NULL) {
     if (!$message->toContactID) {
       // find recipient if $toContactID not set by hook
       if ($message->to) {
-        $message->toContactID = CRM_Core_DAO::singleValueQuery("SELECT contact_id FROM civicrm_phone JOIN civicrm_contact ON civicrm_contact.id = civicrm_phone.contact_id WHERE !civicrm_contact.is_deleted AND phone LIKE %1", [
-          1 => ['%' . $message->to, 'String'],
-        ]);
+        $message->toContactID = Phone::get(FALSE)
+          ->addWhere('contact_id.is_deleted', '=', FALSE)
+          ->addWhere('phone_numeric', 'LIKE', $this->formatPhoneNumber($this->stripPhone($message->to)))
```

Comment:
Incoming SMS phone numbers include country code, which will not be stripped from the value when used in the where query. Will not match on a contacts mobile phone number.